### PR TITLE
Dellaert/issue420

### PR DESCRIPTION
**Pull request by dellaert**
Addresses issue #420

### CODE REVIEW

**Comment by dellaert** [comment]
@mikesheffler tagged you as you frequently find rare problems :-) BTW, MATLAB worked fine on my Mac...

File: gtsam/navigation/PreintegrationBase.h [hunk 1]
```diff
@@ -63,11 +44,6 @@ class GTSAM_EXPORT PreintegrationBase {
   typedef PreintegrationParams Params;
 
  protected:
-  /// Parameters. Declared mutable only for deprecated predict method.
-  /// TODO(frank): make const once deprecated method is removed
-#ifdef GTSAM_ALLOW_DEPRECATED_SINCE_V4
```
Comment (varunagrawal) [inline gtsam/navigation/PreintegrationBase.h:68]: Make params const?

File: gtsam/navigation/PreintegrationBase.cpp [hunk 2]
```diff
@@ -193,22 +193,6 @@ Vector9 PreintegrationBase::computeErrorAndJacobians(const Pose3& pose_i,
   return error;
 }
 
-//------------------------------------------------------------------------------
-#ifdef GTSAM_ALLOW_DEPRECATED_SINCE_V4
-PoseVelocityBias PreintegrationBase::predict(const Pose3& pose_i,
-    const Vector3& vel_i, const imuBias::ConstantBias& bias_i,
-    const Vector3& n_gravity, const Vector3& omegaCoriolis,
-    const bool use2ndOrderCoriolis) const {
-// NOTE(frank): parameters are supposed to be constant, below is only provided for compatibility
```
Comment (varunagrawal) [inline gtsam/navigation/PreintegrationBase.cpp:202]: Make params const?

File: CMakeLists.txt [hunk 3]
```diff
@@ -75,7 +75,7 @@ option(GTSAM_WITH_TBB                    "Use Intel Threaded Building Blocks (TB
 option(GTSAM_WITH_EIGEN_MKL              "Eigen will use Intel MKL if available" OFF)
 option(GTSAM_WITH_EIGEN_MKL_OPENMP       "Eigen, when using Intel MKL, will also use OpenMP for multithreading if available" OFF)
 option(GTSAM_THROW_CHEIRALITY_EXCEPTION  "Throw exception when a triangulated point is behind a camera" ON)
-option(GTSAM_ALLOW_DEPRECATED_SINCE_V4   "Allow use of methods/functions deprecated in GTSAM 4" ON)
+option(GTSAM_ALLOW_DEPRECATED_SINCE_V41  "Allow use of methods/functions deprecated in GTSAM 4.1" ON)
```
[the remark this answers is not in the document]
Comment (dellaert) [reply CMakeLists.txt:78]: No, never a good time to frustrate the hell out of people.

**Comment by dellaert** [comment]
> Make this more descriptive?

"Allow features deprecated in GTSAM 4.1" 

Done

**Comment by dellaert** [comment]
> Make params const? 

Done

File: gtsam/linear/GaussianConditional.h [hunk 4]
```diff
@@ -128,17 +128,9 @@ namespace gtsam {
...
-    /// @name Deprecated
-    /// @{
-    constABlock get_R() const { return R(); }
-    constABlock get_S() const { return S(); }
-    constABlock get_S(const_iterator it) const { return S(it); }
-    const constBVector get_d() const { return d(); }
-    /// @}
-#endif
+
+    // FIXME: deprecated flag doesn't appear to exist?
```
Comment (ProfFan) [inline gtsam/linear/GaussianConditional.h:132]: have you included the `config.h` file?

**Comment by dellaert** [comment]
@step13489mcc see whether you can fix the serialization thing?

**Pull request merged**